app/models/account.rb

```diff
@@ -74,14 +75,22 @@ def switch
     reset!
   end
 
+  # does not use confirm_endpoints! because we can still nil-ify settings
   def reset!
-    solr_endpoint.reset! if solr_endpoint
-    fcrepo_endpoint.reset! if fcrepo_endpoint
-    redis_endpoint.reset! if redis_endpoint
+    SolrEndpoint.reset!
+    FcrepoEndpoint.reset!
+    RedisEndpoint.reset!
   end
 
   private
 
+    # @raise [RuntimeError] if missing any endpoint
+    def confirm_endpoints!
+      raise "Account #{cname} is missing solr_endpoint, cannot switch!" unless solr_endpoint
```

Comment:
I'm just after something a little more specific than a `RuntimeError`. If nothing else, it'll give us a slightly more informative error in logs and honeybadger, and give us something to rescue to present a user-friendly error in the future. I was thinking something generic like `MissingEndpointException`, but if you feel like it's better to have separate ones for every service, 🤷‍♂️ .